annotation_store/tests.py

```diff
@@ -149,14 +157,37 @@ def invalidate(data):
                 else:
                     action()
 
-    def test_lti_passback_triggered_after_create(self):
+    def test_lti_passback_triggered(self):
+        # Test basic calling
         session = self.not_staff_session
         data = object_params_from_session(session)
-        request = create_request(method="post", session=session, data=data)
+        request = create_request(method='post', session=session, data=data)
+        mock_store = mock.create_autospec(AnnotationStore(request, backend_instance=DummyStoreBackend(request)))
+        mock_store.lti_grade_passback()
+        mock_store.lti_grade_passback.assert_called_with()
+
+    # Ensuring no grade passback if not assignment or if teacher
+    @mock.patch.object(ims_lti_py.tool_provider.DjangoToolProvider, 'post_replace_result')
+    def test_lti_passback_not_triggered(self, mock_post_replace_result):
+        session = self.staff_session
+        data = object_params_from_session(session)
+        request = create_request(method='post', session=session, data=data)
         store = AnnotationStore(request, backend_instance=DummyStoreBackend(request))
-        store._lti_grade_passback = mock.Mock(return_value=True)
-        response = store.create()
-        store._lti_grade_passback.assert_called()
+        store.lti_grade_passback()
+        self.assertFalse(mock_post_replace_result.called, "LTI consumer does not expect a grade for the current user and assignment")
+
+    # Test different grades called
+    @mock.patch.object(ims_lti_py.tool_provider.DjangoToolProvider, 'post_replace_result')
+    def test_differing_grades(self, mock_post_replace_result):
+        session = TEST_SESSION_NOT_STAFF
+        data = object_params_from_session(session)
+        request = create_request(method='post', session=session, data=data)
+        store = AnnotationStore(request, backend_instance=DummyStoreBackend(request))
+
+        grades = [.1, .9, .6, 8., .3, .2]
```

Comment:
What about the case when no grade is passed and the default `1.0` value is used? Can we test for that too?

Reply:
we test for no grade in the first function. I added 1.0 as an input, though